libraries/zookeeper_config.rb

```diff
@@ -34,7 +35,11 @@ class ZookeeperConfig < Chef::Resource
       attribute(:properties, option_collector: true, default: {})
 
       def myid
```

Comment:
I would implement this slightly differently. I would set the default value of the instance_id attribute to be the index of the element in the ensemble array. After that we simply add an alias to the method so that `#myid` keeps on working.
```ruby
attribute(:instance_id, kind_of: Integer, default: lazy { ensemble.index(instance_name).next.to_s })
alias_method :myid, :instance_id
```